r18n-core/lib/r18n-core/translated_string.rb

```diff
@@ -59,6 +59,11 @@ def to_s
       end
     end
 
+    # Define `as_json` for ActiveSupport compatibility.
+    def as_json(options = nil)
+      to_str
```

Comment:
Why `to_str` and not `to_s`?

Reply:
Because it definitely returns a String and it's easier to verify/review (given that TranslatedString is a subclass of String and typically only String defines `to_str`). Using `to_s` will probably work too with the current implementation, but historically `to_s` wasn't guaranteed to return a String, and it's more complex to verify/review its behaviour.

Reply:
Please, pay attention, there is redefined `TranslatedString#to_s` method and no `#to_str`.

Maybe you should use `#to_s`, maybe change redefining `#to_s` to `#to_str`.

Reply:
TranslatedString subclasses String, which defines the `to_str` method:

http://ruby-doc.org/core-2.4.2/String.html#method-i-to_str

Reply:
I know, but `#to_s` has `html_safe`, with which the result can be better.

Reply:
A string doesn't need to be marked as html_safe if it's going to be encoding as JSON, ActiveSupport::JSON handles escaping using its own EscapedString class.

Reply:
In `_dump` we use `to_str` method too.

It is good question, do we need `html_safe` in `to_json`.

Anyway, I grant decision to @timcraft :)

Reply:
>A string doesn't need to be marked as html_safe if it's going to be encoding as JSON, ActiveSupport::JSON handles escaping using its own EscapedString class.

OK.

I was also confused by the difference between `as_json` and `to_json`, and projects without ActiveSupport and with system gem `json`, but it's `r18n-rails`, so OK.

Reply:
“as_json and to_json” is a good question

Reply:
Don't misunderstood me, please: I'm not against changes, I'm looking for the best way for everyone :hugs:

Reply:
Oh, no, this is not `r18n-rails`, this is `r18n-core`. Let's be more careful and think about these changes, also for "projects without ActiveSupport"-compatibility.

Reply:
```ruby
> RUBY_VERSION
=> "2.4.2"

> JSON::VERSION
=> "2.0.4"

> 'foo bar'.to_json
=> "\"foo bar\""

> 'foo bar'.as_json
NoMethodError: undefined method `as_json' for "foo bar":String
```

Reply:
I guess that `#as_json` is for ActiveModels, and here we should use `#to_json` with `#to_s` (for `html_safe`.

Reply:
@ai Yes, and `to_str` could also be used instead of `String.new(super)` in the `to_s` method.

@AlexWayfer The as_json method comes from ActiveSupport, and it's compatibility with ActiveSupport that the PR is addressing. If you look at the source of ActiveSupport you can see how it works: to_json calls ActiveSupport::JSON.encode which calls as_json.

Reply:
@timcraft sure. Are we ready to accept this PR? Or you want to replace `String.new(super)`?

Reply:
@AlexWayfer We have `html_safe` things here. The only thing that R18n Core should work without Rails.

Reply:
> The only thing that R18n Core should work without Rails.

OK. Maybe it'll work for Rails and not-Rails with `#to_json` instead of `#as_json`, but I'm not sure.